tests/test_worker.py

```diff
@@ -347,6 +347,16 @@ def test_custom_job_class(self):
         worker = Worker([q], job_class=CustomJob)
         self.assertEqual(worker.job_class, CustomJob)
 
+    def test_custom_job_class_is_not_global(self):
```

Comment:
Perhaps this should be made clear on the documentation, but I'm not sure we need to test this.